pkg/os/copy.go

```diff
@@ -51,9 +52,13 @@ func CopyFileSparse(src, dst string) error {
 	return copyFile(src, dst, true)
 }
 
-func CopySparse(dst io.WriteSeeker, src io.Reader) (int64, error) {
+func CopySparse(ctx context.Context, dst io.WriteSeeker, src io.Reader) (int64, error) {
 	copyBuf := make([]byte, copyChunkSize)
-	sparseWriter := newSparseWriter(dst)
+
+	if ctx == nil {
```

Comment:
Well, I added this before I had found out that allowing context to be nil is bad practice :sweat_smile: and `context.TODO()` should be used instead. But I left the check there just to be sure.

Reply:
If this is a "can never happen" situation, it can be confusing to leave this code which actually is dead code.
If you prefer to keep it for now, can you add a `log.Warnf("ctx is nil, this should not happen")` or similar to all the `ctx == nil` and `ctx != nil` checks?

Reply:
Looking at go code (`exec.CommandContext` implementation), they use
```
	if ctx == nil {
		panic("nil Context")
	}
```

Reply:
Changed to a panic call when context is nil. Thanks for the suggestion.